apps/addons/tests/test_forms.py

```diff
@@ -30,21 +31,45 @@ def setUp(self):
         self.non_existing_name = 'Does Not Exist'
         self.error_msg = 'This name is already in use. Please choose another.'
 
-    def test_new(self):
-        """
-        New add-ons should be able to use non-existing add-on names.
-        """
-        f = forms.AddonForm(dict(name=self.non_existing_name))
-        f.is_valid()
-        eq_(f.errors.get('name'), None)
-
-    def test_new_existing(self):
-        """
-        New add-ons shouldn't be able to use existing add-on names.
-        """
-        f = forms.AddonForm(dict(name=self.existing_name))
-        assert not f.is_valid()
-        eq_(f.errors['name'][0], self.error_msg)
+    def test_update_addon_non_existing_name(self):
```

Comment:
The `AddonForm` is never used in the code to create new addons, it's only used to edit existing ones (from the API). I've thus updated those two tests, and added some new tests.